evap/staff/tests/test_forms.py

```diff
@@ -954,6 +954,29 @@ def test_unused_questionnaire_visibility(self):
         form = EvaluationForm(instance=evaluation, semester=evaluation.course.semester)
         self.assertIn(questionnaire, form.fields["general_questionnaires"].queryset)
 
+    def test_inactive_participants_remain(self):
+        student = baker.make(UserProfile, is_active=False)
+        evaluation = baker.make(Evaluation, course__degrees=[baker.make(Degree)], participants=[student])
+
+        form_data = get_form_data_from_instance(EvaluationForm, evaluation)
+        form = EvaluationForm(form_data, instance=evaluation)
+        self.assertEqual(len(form["participants"]), 1)
+
+    def test_inactive_participants_not_in_queryset(self):
+        evaluation = baker.make(Evaluation, course__degrees=[baker.make(Degree)])
+
+        form_data = get_form_data_from_instance(EvaluationForm, evaluation)
+        form = EvaluationForm(form_data, instance=evaluation)
+        self.assertEqual(len(set(form.fields["participants"].queryset)), 0)
```

Comment:
The querysets should be distinct, so the intermediate sets shouldn't matter. `.count()` should work, so you could use `queryset.count()` instead of `len(set(queryset))`.

Same for the assertions below.